File: php/lib/Ice/LocalExceptions.php
<?php
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

// This file contains all the exception classes derived from LocalException.
// All these exceptions are created from the C++ code.

namespace Ice;

//
// The 6 (7 with the RequestFailedException base class) special local exceptions that can be marshaled in an Ice reply
// message. Other local exceptions can't be marshaled.
//

class RequestFailedException extends LocalException
{
    public Identity $id;
    public string $facet;
    public string $operation;
}

final class ObjectNotExistException extends RequestFailedException {}

final class FacetNotExistException extends RequestFailedException {}

final class OperationNotExistException extends RequestFailedException {}

class UnknownException extends LocalException {}

final class UnknownLocalException extends UnknownException {}

final class UnknownUserException extends UnknownException {}

//
// Protocol exceptions
//

class ProtocolException extends LocalException {}

final class CloseConnectionException extends ProtocolException {}

final class DatagramLimitException extends ProtocolException {}

final class MarshalException extends ProtocolException {}

//
// Timeout exceptions
//

class TimeoutException extends LocalException {}

final class ConnectTimeoutException extends TimeoutException {}

final class CloseTimeoutException extends TimeoutException {}

final class InvocationTimeoutException extends TimeoutException {}

//
// Syscall exceptions
//

class SyscallException extends LocalException {}

final class DNSException extends SyscallException {}

//
// Socket exceptions
//

class SocketException extends SyscallException {}

class ConnectFailedException extends SocketException {}

final class ConnectionLostException extends SocketException {}

final class ConnectionRefusedException extends ConnectFailedException {}

//
// Other leaf local exceptions in alphabetical order.
//

final class AlreadyRegisteredException extends LocalException
{
    public string $kindOfObject;
    public string $id;
}

final class CommunicatorDestroyedException extends LocalException {}

// We don't map closedByApplication because it's very difficult to produce a ConnectionAbortedException or
// ConnectionClosedException without AMI.
final class ConnectionAbortedException extends LocalException {}

final class ConnectionClosedException extends LocalException {}

final class FixedProxyException extends LocalException {}

final class InitializationException extends LocalException {}

final class InvocationCanceledException extends LocalException {}

final class NoEndpointException extends LocalException {}

final class NotRegisteredException extends LocalException
{
    public string $kindOfObject;
    public string $id;
}

final class ParseException extends LocalException {}

final class PluginInitializationException extends LocalException {}

final class SecurityException extends LocalException {}

final class TwowayOnlyException extends LocalException {}

final class PropertyException extends LocalException {}

File: ruby/ruby/Ice/LocalExceptions.rb
# Copyright (c) ZeroC, Inc.

# This file provides all the exceptions derived from Ice::LocalException. Application code should not throw any of
# these exceptions.

module Ice
    #
    # The 7 (8 with the RequestFailedException base class) special local exceptions that can be marshaled in an Ice
    # reply message. Other local exceptions can't be marshaled.
    #

    class DispatchException < LocalException
        def initialize(replyStatus, msg)
            super(msg)
            @replyStatus = replyStatus
        end

        attr_reader :replyStatus
    end

    class RequestFailedException < DispatchException
        def initialize(replyStatus, id, facet, operation, msg)
            super(replyStatus, msg)
            @id = id
            @facet = facet
            @operation = operation
        end

        attr_reader :id, :facet, :operation
    end

    class ObjectNotExistException < RequestFailedException
    end

    class FacetNotExistException < RequestFailedException
    end

    class OperationNotExistException < RequestFailedException
    end

    class UnknownException < DispatchException
    end

    class UnknownLocalException < UnknownException
    end

    class UnknownUserException < UnknownException
    end

    #
    # Protocol exceptions
    #

    class ProtocolException < LocalException
    end

    class CloseConnectionException < ProtocolException
    end

    class DatagramLimitException < ProtocolException
    end

    class MarshalException < ProtocolException
    end

    #
    # Timeout exceptions
    #

    class TimeoutException < LocalException
    end

    class ConnectTimeoutException < TimeoutException
    end

    class CloseTimeoutException < TimeoutException
    end

    class InvocationTimeoutException < TimeoutException
    end

    #
    # Syscall exceptions
    #

    class SyscallException < LocalException
    end

    class DNSException < SyscallException
    end

    #
    # Socket exceptions
    #

    class SocketException < SyscallException
    end

    class ConnectFailedException < SocketException
    end

    class ConnectionLostException < SocketException
    end

    class ConnectionRefusedException < ConnectFailedException
    end

    #
    # Other leaf local exceptions in alphabetical order.
    #

    class AlreadyRegisteredException < LocalException
        def initialize(kindOfObject, id, msg)
            super(msg)
            @kindOfObject = kindOfObject
            @id = id
        end

        attr_reader :kindOfObject, :id
    end

    class CommunicatorDestroyedException < LocalException
    end

    class ConnectionAbortedException < LocalException
    # We don't map closedByApplication because it's very difficult to produce a ConnectionAbortedException or
    # ConnectionClosedException without AMI.
    end

    class ConnectionClosedException < LocalException
    end

    class FeatureNotSupportedException < LocalException
    end

    class FixedProxyException < LocalException
    end

    class InitializationException < LocalException
    end

    class InvocationCanceledException < LocalException
    end

    class NoEndpointException < LocalException
    end

    class NotRegisteredException < LocalException
        def initialize(kindOfObject, id, msg)
            super(msg)
            @kindOfObject = kindOfObject
            @id = id
        end

        attr_reader :kindOfObject, :id
    end

    class ParseException < LocalException
    end

    class PluginInitializationException < LocalException
    end

    class SecurityException < LocalException
    end

    class TwowayOnlyException < LocalException
    end

    class PropertyException < LocalException
    end
end
